### PR TITLE
Setup staging deployment scripts

### PR DESCRIPTION
**Pull request by KPobeeNorris**
We need to migrate from Template Deploy to Kubernetes.  To begin with it has been agreed that we will just move staging, and this PR sets up the initial deployment scripts.

This includes:
* Deployment.yaml
* Ingress.yaml
* Service.yaml
* Kubernetes secrets file
* A new Dockerfile - we are likely going to have dual deployments via both Template Deploy and Kubernetes for a short time and therefore need to have separate Dockerfiles for this period.

### CODE REVIEW

File: Dockerfile.kubernetes [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  && sed -i -e 's/# en_GB.UTF-8 UTF-8/en_GB.UTF-8 UTF-8/' /etc/locale.gen \
+  && dpkg-reconfigure --frontend=noninteractive locales \
+  && update-locale LANG=en_GB.UTF-8
+
+ENV \
+  LANG=en_GB.UTF-8 \
+  LANGUAGE=en_GB.UTF-8 \
+  LC_ALL=en_GB.UTF-8
+
+#ARG VERSION_NUMBER
```
Comment (rossjones) [inline Dockerfile.kubernetes]: Are these not needed?

File: Dockerfile.kubernetes [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    -y \
+    --no-install-recommends \
+    yarn=1.10.1-1 \
+  && yarn add govuk-frontend
+
+COPY Gemfile Gemfile.lock ./
+
+RUN bundle install --without development test --jobs 2 --retry 3
+COPY . /app
+RUN RAILS_ENV=production STAFF_SERVICE_URL=http://example.com SERVICE_URL=http://example.com rails assets:precompile --trace
```
Comment (rossjones) [inline Dockerfile.kubernetes]: Are these env vars overwritten elsewhere?

File: deploy/staging/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        app: prison-visits-public
+    spec:
+      containers:
+      - name: prison-visits-public
+        image: 926803513772.dkr.ecr.eu-west-1.amazonaws.com/prison-visits-booking/prison-visits-public:latest
+        imagePullPolicy: Always
+        command: ['sh', '-c', "bundle exec puma -p 3000 -C ./config/puma_prod.rb --pidfile /tmp/server.pid"]
+        ports:
+        - containerPort: 3000
+#        livenessProbe:
```
Comment (rossjones) [inline deploy/staging/deployment.yaml]: Is the intention to re-add these once the other system is up and running?

File: Dockerfile.kubernetes (hunk 2 above)
Comment (KPobeeNorris) [reply Dockerfile.kubernetes]: We have these ENV VARS listed in the deployment.yaml file so they can be overwritten

**Pull request merged**